### PR TITLE
Remove before and after association callbacks

### PR DESCRIPTION
**Pull request by cheerfulstoic**
This would technically be a breaking change, though I don't know if anybody uses this feature anymore

Pings:
@progm
@subvertallchris

### CODE REVIEW

**Comment by subvertallchris** [comment]
Total support.

**Comment by ProGM** [comment]
I added a couple of notes 👍

**Comment by cheerfulstoic** [comment]
Merged to `master` and I've made a `8.1.x` branch with the changes.  Haven't pushed to 8.1.x because we might want more changes (maybe?)

**Comment by ProGM** [comment]
@cheerfulstoic  Ach, I didn't pressed "submit review", so I think you haven't seen my comments D:

No prob, just a couple of things.
👍  for the 8.1.x branch

File: lib/neo4j/active_node/query/query_proxy.rb [hunk 1]
```diff
@@ -258,6 +254,7 @@ def new_link(node_var = nil)
           self.clone.tap do |new_query_proxy|
             new_query_proxy.instance_variable_set('@result_cache', nil)
             new_query_proxy.instance_variable_set('@node_var', node_var) if node_var
+            new_query_proxy.instance_variable_set('@distinct', @distinct) if @distinct
```
Comment (cheerfulstoic) [inline lib/neo4j/active_node/query/query_proxy.rb]: Good catch, I think it was a bit of code from something that I was playing with.  It wasn't something that was complete so I just removed from the `8.1.x` branch

**Pull request merged**